packages/vats/src/proposals/localchain-test.js

```diff
@@ -0,0 +1,81 @@
+// @ts-check
+import { E } from '@endo/far';
+
+/**
+ * @param {BootstrapPowers & {
+ *   consume: {
+ *     localchain: import('../localchain.js').LocalChain;
+ *   };
+ * }} powers
+ * @param {object} options
+ * @param {{ testResultPath: string }} options.options
+ */
+export const testLocalChain = async (
+  { consume: { chainStorage, localchain } },
+  { options: { testResultPath } },
+) => {
+  console.warn('=== localchain test started (result in', testResultPath, ')!');
+  /** @type {null | ERef<StorageNode>} */
+  let node = await chainStorage;
+  if (node) {
+    for (const nodeName of testResultPath.split('.')) {
+      node = E(node).makeChildNode(nodeName);
+    }
+  }
+
+  let result;
+  try {
+    const lca = await E(localchain).createAccount();
+    console.info('created account', lca);
+    const address = await E(lca).getAddress();
+    console.info('address', address);
+    const balances = await E(lca).allBalances();
+    console.info('balances', balances);
+    await E(lca)
+      .executeTx([
+        {
+          '@type': '/cosmos.bank.v1beta1.MsgSend',
+          from_address: address,
+          to_address: address,
+          amount: [{ denom: 'ucosm', amount: '1' }],
+        },
+      ])
+      .then(
+        res => console.info('unexpected executeTx result', res),
+        e => {
+          if (String(e).match(/insufficient funds/)) {
+            console.info('insufficient funds as expected');
+            return;
+          }
+          console.info('unexpected error', e);
+        },
+      );
+
+    const emptyQuery = await E(localchain).query([]);
+    console.info('emptyQuery', emptyQuery);
+
+    result = { success: true };
+  } catch (e) {
+    console.error('test failed with', e);
+    result = { success: false, error: String(e) };
+  }
+
+  if (node) {
+    await E(node).setValue(JSON.stringify(result));
+  }
+  console.warn('=== localchain test done!', { result });
```

Comment:
If `chainStorage` isn't provided, this test succeeds without testing that. 

Please require `chainStorage` so that this can test integration with it. I think it always is where this proposal is being used.